ospfd/ospf_lsa.c

```diff
@@ -3596,7 +3596,8 @@ int ospf_lsa_different(struct ospf_lsa *l1, struct ospf_lsa *l2,
 	    && CHECK_FLAG((l1->flags ^ l2->flags), OSPF_LSA_RECEIVED))
 		return 1; /* May be a stale LSA in the LSBD */
 
-	assert(l1->size > OSPF_LSA_HEADER_SIZE);
+	if (l1->size == OSPF_LSA_HEADER_SIZE)
```

Comment:
What if the size is < OSPF_LSA_HEADER_SIZE, then it should be an assert condition right? Otherwise the below code will have problem:

memcmp(p1 + OSPF_LSA_HEADER_SIZE, p2 + OSPF_LSA_HEADER_SIZE,
		   l1->size - OSPF_LSA_HEADER_SIZE)

Reply:
I don't believe an assert is *ever* a valid result for a packet received over the wire.  This has taken down many a large network. such formatting errors  should be caught in the parser. (I did not check if it in fact is.)

Reply:
yeah we need to return and log an error in such cases.

Reply:
there may be a corner case not properly covered *ospf_ls_upd_list_lsa, but think this is a separable issue